src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/TimeDerivative.hpp

```diff
@@ -318,7 +318,15 @@ struct ComputeTimeDerivImpl<
           if constexpr (not tmpl::list_contains_v<
                             tmpl::list<GrmhdSourceTags...>, GrmhdDtTags>) {
             // Zero GRMHD tags that don't have sources.
-            dt_var[i] = 0.0;
+            //
+            // Since dt(u) is independent of `dim`, and `dim` is really part
+            // of the `d_dim F^dim` computation, we only need to set a zero
+            // source term when `dim` is zero. This is necessary to avoid `+=`
+            // to a `NaN` (debug mode) or random garbage (release
+            // mode). `add_cartesian_flux_divergence` does a `+=` internally.
+            if (dim == 0) {
+              dt_var[i] = 0.0;
```

Comment:
what was the impact of this bug?

Reply:
For `TildeD` only the `d_z F^z` term of `d_i F^i` was used. No papers have used this code, so there's no impact on science we've published.